packaging/pep517_backend/_backend.py

```diff
@@ -32,12 +31,6 @@ def _chdir_cm(path: os.PathLike) -> t.Iterator[None]:
     get_requires_for_build_sdist as _setuptools_get_requires_for_build_sdist,
 )
 
-with suppress(ImportError):
-    # NOTE: Only available for sdist builds that bundle manpages. Declared by
-    # NOTE: `get_requires_for_build_sdist()` when `--build-manpages` is passed.
-    from docutils.core import publish_file
```

Comment:
Wouldn't it make more sense to avoid `suppress(ImportError)`? If someone e.g. builds the sdist with `python -m build --no-isolation` and docutils isn't available, they'll get a NameError when the code calls the missing function which would be quite confusing.

Reply:
@gotmax23 not at all. If someone chooses not to use pypa/build's built-in virtualenv provisioning, it does not mean they shouldn't provide them. `--no-isolation` only disables the throwaway virtualenv management, putting the responsibility of provisioning them onto the invoker. Besides, `--no-isolation` does not imply `--skip-dependency-check` IIRC, so pypa/build will still call the corresponding hooks to learn which deps are mandatory and check their presence, I think.
But even if you disable both, the code calling these imported bits is only invoked with `--config-setting=--build-manpages`, meaning that `NameError` will only happen under the feature flag — this is when the control flow would actually hit the lines referring to the undefined variables in runtime. Python does not (and cannot) evaluate the references until it reaches them. It'll hit function definitions on import but not their internal scope.

Similarly, if one forgets to install `setuptools`, the same thing is going to happen.

FWIW the pyproject macros should probably provide a way to extract deps needed for building sdists.
The current reusable implementation of calling hooks provides this via https://pyproject-hooks.rtfd.io/en/latest/pyproject_hooks.html#pyproject_hooks.BuildBackendHookCaller.get_requires_for_build_sdist

Reply:
>  meaning that NameError will only happen under the feature flag

Correct.

> FWIW the pyproject macros should probably provide a way to extract deps needed for building sdists.

The macros call `pip wheel` directly and don't build sdists, so they don't need to generate those deps. If you have any feature/bug requests for the pyproject-rpm-macros project, you should [file a Bugzilla](https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&component=pyproject-rpm-macros).

Reply:
> The macros call `pip wheel` directly and don't build sdists, so they don't need to generate those deps.

I know, but they could do so to facilitate your use-case where you're reproducing some of the sdist build processes so you wouldn't have to hardcode the respective deps declaration.